src/core/analysis/mcmc/Mcmcmc.cpp

```diff
@@ -1865,18 +1865,59 @@ void Mcmcmc::tune( void )
 //            std::cout << "chain_heats[" << hotterChainIdx << "]=" << chain_heats[hotterChainIdx] << std::endl;
         }
         
-        // if the heat of a given hot chain is smaller than the minimum bound
-        // interpolate this heat and the heats of all the hotter chains
-        // to fall between the lowest heat that is greater than the minimum bound and the minimum bound
-        if (j < num_chains)
+        /* If the heat of a given hot chain is smaller than the minimum bound, we want to linearly interpolate
+         * this heat and the heats of all the hotter chains to fall between the lowest heat that is greater
+         * than the minimum bound and the minimum bound.
+         * The formula for linearly interpolating between (x1,y1) and (x2,y2) is y(x) = y2 - (y2-y1)*[ (x-x1)/(x2-x1) ].
+         *
+         * Let "heat(i)" be a notational shortcut for chain_heats[ chainForHeatIndex(i) ], and let k be the heat
+         * index of the coldest of such "bad" chains. Then, its heat can be calculated as
+         *
+         * log(heat(k)) = log(heat(k-1)) - [ log(heat(k-1)) - log(heatMinBound) ]*[ (k - (k-1)) / (num_chains - (k-1)) ]
+         *              = log(heat(k-1)) - [ log(heat(k-1)) - log(heatMinBound) ]*[ 1 / (num_chains - (k-1)) ]
+         *
+         * Let us further denote heat(k-1) as "lowestGreaterThanMin", and k-1 as "m". Then,
+         *
+         * log(heat(k)) = log(lowestGreaterThanMin) - [ log(lowestGreaterThanMin) - log(heatMinBound) ]*[ 1 / (num_chains - m) ]
```

Comment:
Is `k` a variable here?  If so, it seems like there should be a `k` on the right-hand-side.  So then maybe `k` is not a variable?